icu4c/source/common/locid.cpp

```diff
@@ -505,8 +501,11 @@ Locale& Locale::operator=(Locale&& other) U_NOEXCEPT {
     variantBegin = other.variantBegin;
     fIsBogus = other.fIsBogus;
 
-    other.baseName = other.fullName = other.fullNameBuffer;
-    other.setToBogus();
+    if (other.fullName != other.fullNameBuffer ||
+        other.baseName != other.fullName) {
```

Comment:
Hard to read due to same indentation for condition & body. Please merge this line with the previous, or indent this line by another 4 spaces. Or look at my suggested alternative.

Reply:
The very first version of this code would be a good way to reach the goal of doing as little unnecessary work as possible, just making sure that the destructor of `other` won't try to deallocate anything it shouldn't but nothing more. I've reverted to that version now, it's by far the simplest implementation we've looked at so far.

One might even delete the `other.fIsBogus = TRUE` as well, as we (as you point out) don't expect the `other` object to be touched between the point where it's moved and the point where it's destructed. I mostly added that line as a little "sugar", which makes it easy to unit test that this part of the code has been reached.